### PR TITLE
Add metadata to attributes

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it adds metadata editor to attributes.
Build on top of #669 

**PR intended to be tested with API branch:** master

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [CU-6a5v1n POST-MVP: Allow user to create and edit Metadata for Attributes](https://app.clickup.com/t/6a5v1n)

File: src/attributes/views/AttributeCreate/AttributeCreate.tsx [hunk 1]
```diff
@@ -104,87 +106,83 @@ const AttributeDetails: React.FC<AttributeDetailsProps> = ({ params }) => {
...
-                    values: values.map(value => ({
-                      name: value.name
-                    }))
-                  }
-                }
-              })
+    <>
+      <AttributePage
+        attribute={null}
+        disabled={false}
```
Comment (krzysztofwolski) [inline src/attributes/views/AttributeCreate/AttributeCreate.tsx:112]: Shouldnt we block from when data is loading?

File: src/fragments/metadata.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import gql from "graphql-tag";
+
+export const metadataFragment = gql`
+  fragment MetadataItem on MetadataItem {
+    key
+    value
+  }
+  fragment Metadata on ObjectWithMetadata {
```
Comment (krzysztofwolski) [inline src/fragments/metadata.ts:8]:
```suggestion
  fragment MetadataFragment on ObjectWithMetadata {
```

File: src/fragments/products.ts [hunk 3]
```diff
@@ -105,8 +106,10 @@ export const productFragmentDetails = gql`
   ${productVariantAttributesFragment}
   ${stockFragment}
   ${weightFragment}
+  ${metadataFragment}
   fragment Product on Product {
     ...ProductVariantAttributesFragment
+    ...Metadata
```
Comment (krzysztofwolski) [inline src/fragments/products.ts:112]:
```suggestion
    ...MetadataFragment
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: src/attributes/components/AttributePage/AttributePage.tsx [hunk 4]
```diff
@@ -102,66 +118,84 @@ const AttributePage: React.FC<AttributePageProps> = ({
...
+              {intl.formatMessage(sectionNames.attributes)}
+            </AppHeader>
+            <PageHeader
+              title={
+                attribute === null
+                  ? intl.formatMessage({
+                      defaultMessage: "Create New Attribute",
+                      description: "page title"
+                    })
+                  : maybe(() => attribute.name)
```
Comment (AlicjaSzu) [inline src/attributes/components/AttributePage/AttributePage.tsx:152]: `attribute?.name`

**Review by AlicjaSzu — APPROVED** [review]

File: src/fragments/metadata.ts (hunk 2 above)
Comment (dominik-zeglen) [reply src/fragments/metadata.ts:8]: I'll change it when all PRs would be merged, so I'll be doing it once

File: src/attributes/views/AttributeCreate/AttributeCreate.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply src/attributes/views/AttributeCreate/AttributeCreate.tsx:112]: Probably

**Review by orzechdev — APPROVED** [review]

**Pull request merged**